openshift/ocp-release-operator-sdk
# Merge upstream tag v1.25.1

Operator SDK v1.25.1

Merge executed via ./UPSTREAM-MERGE.sh v1.25.1

**Note**
The only change that was made that will impact users is an update to the `operator-sdk bundle validate` logic that will now verify that no potentially deprecated APIs are found in the `ClusterServiceVersion`.

To test this functionality:
1. I cloned the community-operators repo with:
```shell
git clone https://github.com/k8s-operatorhub/community-operators.git
```
2. I navigated to the couchbase-enterprise operator directory:
```shell
cd community-operators/operators/couchbase-enterprise
```
3. Ran:
```shell
 operator-sdk bundle validate 2.3.0/ --select-optional suite=operatorframework --optional-values=k8s-version=1.25
```

4. Verified the output matched the expected of:
```
WARN[0000] Warning: Value couchbase-operator.v2.3.0-1: this bundle is using APIs which were deprecated and removed in v1.25. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-25. Migrate the API(s) for events: (["ClusterServiceVersion.Spec.InstallStrategy.StrategySpec.Permissions[0].Rules[8]"]),poddisruptionbudgets: (["ClusterServiceVersion.Spec.InstallStrategy.StrategySpec.Permissions[0].Rules[10]"]), 
ERRO[0000] Error: Value couchbase-operator.v2.3.0-1: this bundle is using APIs which were deprecated and removed in v1.25. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-25. Migrate the API(s) for CronJob: (["ClusterServiceVersion.Spec.CustomResourceDefinitions.Owned[8].Resource[0]"]) 
WARN[0000] Warning: Value couchbase-operator.v2.3.0-1: this bundle is using APIs which were deprecated and removed in v1.25. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-25. Migrate the API(s) for events: (["ClusterServiceVersion.Spec.InstallStrategy.StrategySpec.Permissions[0].Rules[8]"]),poddisruptionbudgets: (["ClusterServiceVersion.Spec.InstallStrategy.StrategySpec.Permissions[0].Rules[10]"]), 
ERRO[0000] Error: Value couchbase-operator.v2.3.0-1: this bundle is using APIs which were deprecated and removed in v1.25. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-25. Migrate the API(s) for CronJob: (["ClusterServiceVersion.Spec.CustomResourceDefinitions.Owned[8].Resource[0]"]) 
WARN[0000] Warning: Value couchbase-operator.v2.3.0-1: unable to find the resource requests for the container: (couchbase-operator). It is recommended to ensure the resource request for CPU and Memory. Be aware that for some clusters configurations it is required to specify requests or limits for those values. Otherwise, the system or quota may reject Pod creation. More info: https://master.sdk.operatorframework.io/docs/best-practices/managing-resources/ 
```
